### ISSUE TITLE
Unable to launch AltStore BETA

### ISSUE THREAD

**Issue by Sviek** [rileytestut/AltStore#1456]
Issue is only with the beta version. Had the issue initially a handful of times, closing and trying to run it, opening it from AltStore. It opened fine once inbetween that, and I was able to start downloading Delta beta, then I closed AltStore beta and it started the error again.

iPhone 11
v17.4.1

Unable to Launch AltStore
Error
Domain=NSCocoaError-Domain Code=134110 "An error occurred during persistent store migration."
UserInfo={sourceURL=file: ///private/var/mobile/ Containers/Shared/ AppGroup/
966B8D31-0E5B-4C13- BFF1-8DF9F7D6787F/ Database/AltStore.sqlite, reason=Cannot migrate
store in-place: Validation error missing attribute values on mandatory destination attribute, destinationURL=file:/// private/var/mobile/ Containers/Shared/
AppGroup/
966B8D31-0E5B-4C13-
BFF1-8DF9F7D6787F/
Database/AltStore.sqlite, NSUnderlyingError=0x303 534000 {Error
Domain=NSCocoaError-
Domain Code=134110 "An
error occurred during persistent store migration."
UserInfo={entity=AppPer- mission, attribute=appBundleID, reason=Validation error missing attribute values on mandatory destination attribute}}}
Error
Domain=NSCocoaError- Domain Code=134110 "An error occurred during persistent store migration."
UserInfo={entity=AppPer- mission, attribute=appBundleID, r..."

**Comment by markpavlenko** [comment]
same

**Comment by alexchandel** [comment]
I've had this issue for a year+.  Seems like they upgraded their sqlite db improperly and permanently broke it for unlucky users. @rileytestut Can you please push a workaround for this?

**Comment by alexchandel** [comment]
The fix is to delete both regular AltStore and AltStore beta.  Having an older AltStore somehow stops the newer Beta from working.

**Comment by mrowlinson** [comment]
> The fix is to delete both regular AltStore and AltStore beta. Having an older AltStore somehow stops the newer Beta from working.

This works for me once. The beta app will launch immediately after install and then will get the above error every time after that.

**Comment by rebelancap** [comment]
Any fix for this? My beta version (really should be called the paid version) keeps getting this issue. The only reason I'm paying for the beta is to have custom repositories. And it rarely works. Feel like I need to delete and reinstall the beta constantly. 

Also, I'm afraid of deleting the non-beta because the beta keeps breaking. And the non-beta has duplicate sources when you have both installed. It's a bit of a mess all around.

**Comment by mrowlinson** [comment]
Yeah, once you install the beta version you have to delete the prod version before opening the beta.On Aug 24, 2024, at 9:30 AM, iamse7en ***@***.***> wrote:﻿
Any fix for this? My beta version (really should be called the paid version) keeps getting this issue. The only reason I'm paying for the beta is to have custom repositories. And it rarely works.
